### DISCUSSION TITLE
Bringing Back "Teletype" Feature in Atom to Pulsar

### DISCUSSION THREAD

**Discussion by huybin1234b** [pulsar-edit/.github#236] [Package Ideas]
Hey everyone! 👋 I'm diving into the world of coding and development, and I've got a cool idea to share with you all. I've been exploring Atom, and one feature that caught my attention was "Teletype." It used to be a part of Atom, allowing collaborative coding in real-time, which sounds amazing for learning and working together, right?

Now, here's the thing: I'm a bit new to this coding scene, and honestly, I'm still learning the ropes. But I thought, "Hey, why not suggest something cool?" So, I'm proposing to bring back the "Teletype" feature through a package for Pulsar.

I might not know all the ins and outs yet, but I've heard about markdown and figured out how to use it here (yay, progress!). And yes, I'm also using some AI magic to help me craft this proposal because, well, I'm a newbie here!

Imagine how awesome it would be to have Teletype back in action. It could help beginners like me learn from more experienced devs, collaborate with peers, and build some amazing stuff together. Plus, it could add a whole new dimension to the Pulsar environment!

I'm really excited about the possibilities, and I'd love to hear your thoughts and suggestions. Let's make coding more collaborative and fun for everyone! 😊✨

Feel free to drop in your ideas and expertise—I'm all ears (well, metaphorically speaking, of course)!

**Comment by Daeraxa** [comment]
This comes up every so often and generally, yeah, we are pretty keen for this to come back. The problem is how to do it, how much of the original Teletype package can be re-used etc.

You might like to have a look at https://github.com/pulsar-edit/pulsar/issues/536 where most of this discussion has taken place so far.

**Reply by confused-Techie** [reply]
@huybin1234b Reading the linked discussion should provide all the context and information on this subject as we have currently.

I've tried a few times to tackle the Teletype problem without much success beyond a large time investment or a complete reworking of the tool conceptually and practically. But would be happy to work to find a solution to the issue
